### ISSUE TITLE
SCM-TFVC: Unable to annotate the project which TF400324

### ISSUE THREAD

**Issue by ntwbrown** [lg2de/sonar-scm-tfvc#57]
Hello, I am using **SonarQube Community Edition Version 8.2** (build 32929)
T**FS 2017 Version 1** (on prem)
with **SonarQube TFVC Plugin 2.3** (sonar-scm-tfvc-plugin-2.3.0-73)

I have configured TFVC as at global level Administration > Configuration > SCM > TFVC with PersonalAccessToken (with sufficient rights) and **CollectionURI**

Within the ‘Prepare the SonarQube analysis’ Build step, I have added the following **/d:sonar.scm.enabled=true /d:sonar.scm.provider=tfvc**

However during the ‘Complete the SonarQube Analysis’ stage I get the following error…

`SCM-TFVC: Unable to annotate the project which TF400324: Azure DevOps services are not available from server https://servername/tfs/DefaultCollection`

**Comment by DEYS** [comment]
I'm getting the same issue.  I'm using the same plugin version but TFS 2019

![image](https://user-images.githubusercontent.com/12831058/79496810-ada15280-7ff4-11ea-84f6-9b8f41b2315e.png)

**Comment by venudhar1** [comment]
I'm getting the same issue with almost same configuration as above. Did anyone got the solution for the same. Appreciate if anyone could share the solution

**Comment by azharbroadpeak** [comment]
I am getting the same error.

**Comment by lg2de** [comment]
Are there more information available in the analyzer output? Please provide as much details as possible from "begin analysis" and "end analysis" including command line options and analyzer output.

So far I'm using SQ 8.7 and plugin 2.3.0 with TFS/DevOps server 2019 without such issues.

Sorry for strong delay for my response: I cannot remember to have received mails for all these comments above...

**Comment by muhyid** [comment]
Dear Lukas, @lg2de 
I also have the same issue. I will try to explain the details in the following

My components:
SonarQube Server: Community Edition Version 8.8 (build 42792)
SonarScanner for .NET: sonar-scanner-msbuild-5.2.0.29862-net46
SonarScanner for Azure DevOps: 4.19.0
Azure DevOps Server Version Dev17.M153.5 (On-Premise)
TFVC Plugin: sonar-scm-tfvc-plugin-2.3.0-73
Windows Server 2016 (latest patch level)

1.	I have configured the TFVC plugin on the SonarQube Server with the username and password

![image](https://user-images.githubusercontent.com/83450450/116708989-eb430480-a9d8-11eb-9ce2-1a2dfe8421d6.png)
 
2.	I tried to analyze my .NET codes with two methods. 
2.1.	**With SonarScanner for .NET (.NET Framework 4.6+)**
First 2 phases (SonarScanner.MSBuild begin and msbuild) there was no problem.
![image](https://user-images.githubusercontent.com/83450450/116710865-e7b07d00-a9da-11eb-9302-8b7c014a0abc.png)

I encountered the problem during the last (end) phase.
SonarScanner.MSBuild end:
00:55:05.158 INFO: SCM Publisher SCM provider for this project is: tfvc
00:55:05.158 INFO: SCM Publisher 1360 source files to be analyzed
00:55:05.173 DEBUG: SCM-TFVC: Executing the TFVC annotate command: C:\sonar-scanner-msbuild\.sonarqube\out\.sonar\.sonartmp\SonarTfsAnnotate8600626928480430280.exe
00:55:05.815 DEBUG: SCM-TFVC: received output: <Enter your credentials>
00:55:05.815 DEBUG: SCM-TFVC: received output: <Enter the Collection URI>
00:55:06.378 INFO: SCM Publisher 0/1360 source files have been analyzed (done) | time=1205ms
00:55:06.378 ERROR: SCM-TFVC: Unable to annotate the project which TF400324: Azure DevOps services are not available from server https://AzureDevOps.mycompany.com/CollectionName.
00:55:06.393 WARN: Missing blame information for the following files:
00:55:06.394 WARN:   * AAA/BBB/CCC/DDD/EEE/FFF.html
00:55:06.394 WARN:   * GGG/HHH/III/JJJ/KKK/LLL.cs
00:55:06.394 WARN:   * MMM/NNN/OOO/PPP/RRR.cs
00:55:06.394 WARN:   * AAA/SSS/TTT/UUU/VVV/YYY.ts
…
…
00:55:07.182 WARN: This may lead to missing/broken features in SonarQube
2.2.	**With SonarScanner for Azure DevOps**
There was no problem with the Prepare Analysis Configuration step
Additional Properties:
sonar.tfvc.collectionuri= https://AzureDevOps.mycompany.com/CollectionName/
I encountered the same problem as the standalone scanner during the Run Code Analysis step
Run Code Analysis step:
19:39:54.126 INFO: SCM Publisher SCM provider for this project is: tfvc
19:39:54.126 INFO: SCM Publisher 1360 source files to be analyzed
19:39:54.142 DEBUG: SCM-TFVC: Executing the TFVC annotate command: C:\Agent\a0\_work\1\.sonarqube\out\.sonar\.sonartmp\SonarTfsAnnotate2626423733292379477.exe
19:39:55.003 DEBUG: SCM-TFVC: received output: <Enter your credentials>
19:39:55.003 DEBUG: SCM-TFVC: received output: <Enter the Collection URI>
##[error]19:39:55.599 ERROR: SCM-TFVC: Unable to annotate the project which TF400324: Azure DevOps services are not available from server https://AzureDevOps.mycompany.com/CollectionName/.
##[debug]Processed: ##vso[task.logissue type=error;]19:39:55.599 ERROR: SCM-TFVC: Unable to annotate the project which TF400324: Azure DevOps services are not available from server https://AzureDevOps.mycompany.com/CollectionName/.
19:39:55.599 ERROR: SCM-TFVC: Unable to annotate the project which TF400324: Azure DevOps services are not available from server https://AzureDevOps.mycompany.com/CollectionName/.
19:39:55.599 INFO: SCM Publisher 0/1360 source files have been analyzed (done) | time=1457ms
19:39:55.599 WARN: Missing blame information for the following files:
19:39:55.599 WARN:   * AAA/BBB/CCC/DDD/EEE/FFF/GGG.cs
19:39:55.599 WARN:   * AAA/BBB/CCC/HHH/III.cs
19:39:55.599 WARN:   * AAA/JJJ/KKK/LLL/MMM/NNN.ts
19:39:55.599 WARN:   * AAA/OOO/PPP/RRR/SSS/TTT.xml
…
…
19:39:55.692 WARN: This may lead to missing/broken features in SonarQube

3.	My Azure DevOps Server is working properly. There are so many build pipelines running on it. If I disable the SCM (sonar.scm.disabled=true), both methods are completing the analysis without any errors.
Do you have any comments?
P.S: I masked file and folder names in the logs

**Comment by lg2de** [comment]
Sorry, but so far I've no idea what "TF400324" wants to tell us.

I found several post in the net which are unrelated to this plugin, but mentioning TF400324.
They advice to clear the cache on the agent. Would you please try to clear the cache and retry?

Additionally I've started a new PR with extended logging details.
Could you please retry with the testing version at https://ci.appveyor.com/api/buildjobs/8wfybh4qxs08b95w/artifacts/sonar-scm-tfvc-plugin%2Ftarget%2Fsonar-scm-tfvc-plugin-2.3.1.113.jar

Additional question: Since I've taken ownership of this project I always use "Personal Access Token" instead of user name and password. Could you please check whether this could be an option for you and maybe it solves the problem (somehow).
Next week I'll try to test the configuration with user name and password in my test system.

**Comment by lg2de** [comment]
I've tested the plugin with using "user name and password" instead of "PAT". This is working fine.
So, I've currently no idea to get a solution for your problem.

**Comment by azharbroadpeak** [comment]
@lg2de: can you share the screen shot of annotation that is working with you and exact user name in sonar qube and scm ? and also your sonarqube is integrated with active directory ? both users are of same active directory?

**Comment by azharbroadpeak** [comment]
by they way annotation is now working fine with 8.3.1.34397 free community version.

**Comment by azharbroadpeak** [comment]
does issues are being assign to last commiter automatically?

**Comment by lg2de** [comment]
> @lg2de: can you share the screen shot of annotation that is working with you and exact user name in sonar qube and scm ? and also your sonarqube is integrated with active directory ? both users are of same active directory?

Here is a screeshot of the configuration and the analysis. But, I think you will not get useful input. What are you looking for?
![image](https://user-images.githubusercontent.com/44983012/118658740-f37aac80-b7ec-11eb-8bbb-da16a0e86b9b.png)
![image](https://user-images.githubusercontent.com/44983012/118658502-bc0c0000-b7ec-11eb-9a6c-36c08c022458.png)

The sonarqube server is NOT integrated directly with the active directory, because it is running on Linux.
The user authentication is integrated with the active directory.
Any yes, currently we have to domains and it is working fine with both.

> by they way annotation is now working fine with 8.3.1.34397 free community version.

Fine.

> Do issues are being assign to last commiter automatically?

I think, this question is out of scope for current issue.
Please open a new issue with detailed description of your question.

**Comment by azharbroadpeak** [comment]
sure thanks.🙂

**Comment by lg2de** [comment]
Closing the issue for the moment. Feel free to update with more details.

**Comment by AliaksandrTseleshTR** [comment]
Hello! Looks like I am having a similar issue. We are using Azure DevOps Services with SonarQube extension v.5.4.0 and SonarQube 9.3 with sonar-scm-tfvc plugin v.2.3.1.

I'm passing these parameters during the 'Prepare analysis on SonarQube' step:
sonar.scm.provider=tfvc
sonar.tfvc.collectionuri=$(System.CollectionUri)
sonar.tfvc.pat.secured=$(ADO_PAT)

But getting this error during the execution:
![image](https://user-images.githubusercontent.com/92371797/170242649-d1018cbd-fa72-4dc9-acc7-10074a1e1ed2.png)

I've tried to copy SonarTfsAnnotate.exe from the build controller and run it passing PAT and URI manually, but getting the same error:
![image](https://user-images.githubusercontent.com/92371797/170243417-4dc96df5-99ec-4cd2-9e41-30504bdc3210.png)

**Comment by lg2de** [comment]
@AliaksandrTseleshTR, which privileges/"Scopes" did you assigned to the generated PAT?
I'm using `Code (Full)` and `Identity (Read & manage)`.

**Comment by lg2de** [comment]
Sorry, but I have still no idea what is wrong here.
In the past I tested successfully with DevOps services.

**Comment by AliaksandrTseleshTR** [comment]
@lg2de Looks like I found the root cause of the issue. Microsoft deprecated usage of TLS 1.0/1.1 with Azure DevOps on March 31, 2022. https://devblogs.microsoft.com/visualstudio/azure-devops-requires-tls-1-2-on-all-connections-including-visual-studio/
I created a few registry keys to enable TLS 1.2 for all connections forcible, it works for me. But, I think, it would be better to update the plugin code to do it on the plugin side.
